ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -2613,16 +2475,18 @@ static void ggml_sycl_op_mul_mat(ggml_backend_sycl_context & ctx, const ggml_ten
                     GGML_ABORT("fatal error");
                 }
 
-                if (convert_src1_to_q8_1 && !src1_is_contiguous) {
-                    scope_op_debug_print scope_dbg_print(__func__, "/quantize_row_q8_1_sycl", dst,
-                                                         /*num_src=*/2, " : converting src1 to Q8_1");
-                    quantize_row_q8_1_sycl(src1_ddf_i, src1_ddq_i, ne10, src1_ncols, src1_padded_col_size, false, stream);
-                    /*
-                    DPCT1010:92: SYCL uses exceptions to report errors and does
-                    not use the error codes. The call was replaced with 0. You
-                    need to rewrite this code.
-                    */
-                    SYCL_CHECK(0);
+                if constexpr(quantize_enabled) {
+                    if (!src1_is_contiguous) {
```

Comment:
Since you're changing this I think there could be another useful improvement. It always confused me why this is a separate `if` block from the above.
I think what we want here is something like:
```
if constexpr(quantize_enabled) {
  if (src1_is_contiguous) {
    [...]
  } else {
    [...]
  }
}
```

Reply:
That looks better thanks!